### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by charliepoker**

### CODE REVIEW

File: app/config/auth.config.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  secret: "bezkoder-secret-key"
```
Comment (ozombo) [inline app/config/auth.config.js:2]: Why is your secret key this simple and the same from the tutorial?

File: app/controllers/auth.controller.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+          accessToken: token,
+        });
+      });
+    })
+    .catch((err) => {
+      res.status(500).send({ message: err.message });
+    });
+};
+
+exports.verifyOtp = (req, res, next) => {
```
Comment (ozombo) [inline app/controllers/auth.controller.js:116]: How are you handling an already verified OTP?

File: app/controllers/auth.controller.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        });
+      } else {
+        user
+          .update({
+            isVerified: true,
+          })
+          .then(
+            res.status(200).json({
+              message: "Verification Complete",
+              data: user,
```
Comment (ozombo) [inline app/controllers/auth.controller.js:137]: This returns all user data including the hashed password I am certain, why?

File: app/controllers/user.controller.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+exports.allAccess = (req, res) => {
+  res.status(200).send("Public Content.");
+};
+
+exports.userBoard = (req, res) => {
+  res.status(200).send("User Content.");
```
Comment (ozombo) [inline app/controllers/user.controller.js:6]: How about sending the user data instead of this static json?

File: app/middleware/verifySignUp.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+const db = require("../models");
+const ROLES = db.ROLES;
+const User = db.user;
+
+checkDuplicateUsernameOrEmail = (req, res, next) => {
+  // Username
+  User.findOne({
```
Comment (ozombo) [inline app/middleware/verifySignUp.js:7]: Unify this to find the username, email, and phone number at once. Running three functions that could easily be combined into one is bad performmance.

File: app/models/user.model.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    phoneNumber: {
+      type: Sequelize.BIGINT,
+      unique: true,
+    },
+    accountNumber: {
+      type: Sequelize.BIGINT,
+      unique: true,
+      allowNull: false,
+    },
+    otp: {
```
Comment (ozombo) [inline app/models/user.model.js:38]: How are errors handled in the case of an already existing OTP which isn't unlikely.

File: app/routes/user.routes.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+module.exports = function(app) {
+  app.use(function(req, res, next) {
+    res.header(
+      "Access-Control-Allow-Headers",
+      "x-access-token, Origin, Content-Type, Accept"
+    );
+    next();
+  });
+
+  app.get("/api/test/all", controller.allAccess);
```
Comment (ozombo) [inline app/routes/user.routes.js:13]: Why are the endpoints named test?

File: server.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+// database
+const db = require("./app/models");
+const Role = db.role;
+
+// db.sequelize.sync();
+// force: true will drop the table if it already exists
+db.sequelize.sync({ force: false}).then(() => {
+  console.log("Drop and Resync Database with { force: true }");
+  // initial();
```
Comment (ozombo) [inline server.js:23]: Write a function to run this based on an env config. So if its dev run it but if PROD don't run. Always understand you have to figure out to run this once upon deployment else the roles won't be created.

**Review by ozombo — CHANGES_REQUESTED** [review]
Why are the JWT flow images added to this commit?

File: app/config/auth.config.js (hunk 1 above)
Comment (charliepoker) [reply app/config/auth.config.js:2]: fixed

File: app/controllers/auth.controller.js (hunk 3 above)
Comment (charliepoker) [reply app/controllers/auth.controller.js:137]: it returns only user fullname now

File: app/config/auth.config.js (hunk 1 above)
Comment (ozombo) [reply app/config/auth.config.js:2]: Where is the commit? The pr isn't updated

File: app/controllers/auth.controller.js (hunk 3 above)
Comment (ozombo) [reply app/controllers/auth.controller.js:137]: Commit?

Comment (ozombo) [reply app/controllers/auth.controller.js:137]: Also why are you returning only the full name when other vital info would be needed? @charliepoker

File: app/config/auth.config.js (hunk 1 above)
Comment (charliepoker) [reply app/config/auth.config.js]: I have modified the secret key

File: app/controllers/auth.controller.js (hunk 3 above)
Comment (charliepoker) [reply app/controllers/auth.controller.js]: I have modified the returned data from the user.

File: app/routes/user.routes.js (hunk 7 above)
Comment (charliepoker) [reply app/routes/user.routes.js]: they have been named correctly

File: app/models/user.model.js (hunk 6 above)
Comment (charliepoker) [reply app/models/user.model.js:38]: errors are been handled now

File: app/controllers/auth.controller.js (hunk 2 above)
Comment (charliepoker) [reply app/controllers/auth.controller.js:120]: it has been handled

**Review by ozombo — APPROVED** [review]

**Pull request merged**